public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -334,7 +334,7 @@ export class TimeSrv {
       range.from = range.from.valueOf().toString();
     }
     if (isDateTime(range.to)) {
-      range.to = range.to.valueOf().toString();
+      range.to = rangeUtil.adjustDateTimeToMaxMillis(range.to);
```

Comment:
so the idea here is to take into account "the whole second", since there could be data points inside that range?